### PR TITLE
filter: don't record tombstone store in filter metrics

### LINKED ISSUE #5386: filter metrics should not record the tombstone store
## Enhancement Task
If we have many tombstone stores, there is a performance issue in both Grafana and PD itself. Since these tombstone stores have been removed completely, it doesn't matter if we don't record them. Furthermore, we can exclude the store with the tombstone state when getting them.

### PR DESCRIPTION
**Pull request by rleungx**
Signed-off-by: Ryan Leung <rleungx@gmail.com>

### What problem does this PR solve?

Issue Number: Close #5386.

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Manual test (add detailed scripts or steps below)

Before:
Screen Shot 2022-08-02 at 1 37 48 PM
After:
Screen Shot 2022-08-02 at 1 41 24 PM

### Release note

```release-note
None
```

### CODE REVIEW

**Review by JmPotato — COMMENTED** [review]
Unnecessary change?

image

**Comment by rleungx** [comment]
> Unnecessary change?
> 
> image

The previous commit modify the permission, I just change it back.

File: server/schedule/filter/filters.go [hunk 1]
```diff
@@ -34,7 +34,8 @@ import (
 func SelectSourceStores(stores []*core.StoreInfo, filters []Filter, opt *config.PersistOptions) []*core.StoreInfo {
 	return filterStoresBy(stores, func(s *core.StoreInfo) bool {
 		return slice.AllOf(filters, func(i int) bool {
-			if !filters[i].Source(opt, s).IsOK() {
+			status := filters[i].Source(opt, s)
+			if !status.IsOK() && status != statusStoreTombstone {
```
Comment (lhy1024) [inline server/schedule/filter/filters.go:38]: Maybe we can add 'not tombstone' to `status.IsOK()`?

Comment (rleungx) [reply server/schedule/filter/filters.go]: I think it belongs to the different types. The previous change has a problem, I have updated it.

**Review by nolouch — APPROVED** [review]

**Comment by nolouch** [comment]
PTAL @CabinfeverB

**Review by CabinfeverB — APPROVED** [review]

**Pull request merged**